### PR TITLE
Feature/icns names as nfts

### PR DESCRIPTION
**Pull request by tomiir**
## Changelog

- Adds new controller with ICNS adapter
- ICNS names now come as an NFT collection.
- Adds ICNSDisplay to display the NFT images both in the list and the details
- Refactors NFT screen to accomodate ICNS names.
- Fixes overflow issue on nft screen

### Type of changes included:

- [X] Components
- [X] Business Logic
- [ ] Bug fixes
- [ ] Styling
- [ ] Code Refactor

### Clubhouse Tickets Related:

- https://www.notion.so/Plug-Extension-631a299f3ec14b9c9522e29a9800e923?p=e471b6fc34464baaa4456c89b5202440

### Screenshots/Gifs:

image

https://user-images.githubusercontent.com/84542297/164484274-4686c34b-122b-40fc-9c23-1550fa788447.mov

### Tested on:

- [X] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/ui/ICNSDisplay/index.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <div className={`${className} ${classes.wrapper} ${loading ? classes.loadingWrapper : ''}`}>
+      <img
+        className={classes.icnsBackground}
+        onClick={onClick}
+        onLoad={() => setLoading(false)}
+        src={icns.url}
+      />
+      <span className={classes.icnsName}>
+        {icns.name?.length > 12 ? shortAddress(icns?.name, 3, 6) : icns?.name}
```
Comment (riveign) [inline source/ui/ICNSDisplay/index.jsx:25]: this magic numbers le, should we use a constant for this?

**Review by riveign — APPROVED** [review]
Small comment

File: source/ui/ICNSDisplay/index.jsx (hunk 1 above)
Comment (rocky-fleek) [reply source/ui/ICNSDisplay/index.jsx:25]: good catch!

File: source/components/NFTs/styles.js [hunk 2]
```diff
@@ -47,7 +50,7 @@ export default makeStyles((theme) => ({
     justifyContent: 'center',
     padding: `${theme.spacing(1)}px ${theme.spacing(4)}px`,
     textAlign: 'center',
-    height: 320 + 118,
+    height: EXTENSION_HEIGHT,
```
Comment (riveign) [inline source/components/NFTs/styles.js:53]: nice

**Review by angarita-dev — APPROVED** [review]

**Pull request merged**